cypress/integration/pages/mediaAssetPage/testsForAMPOnly.js

```diff
@@ -14,35 +13,43 @@ export const testsThatFollowSmokeTestConfigForAMPOnly = ({
   service,
   pageType,
   variant,
-}) =>
+}) => {
   describe(`testsThatFollowSmokeTestConfigForAMPOnly for ${service} ${pageType}`, () => {
     describe('Media Player', () => {
       const language = appConfig[config[service].name][variant].lang;
-      let embedUrl;
-
-      beforeEach(() => {
-        cy.request(`${Cypress.env('currentPath')}.json`).then(({ body }) => {
-          embedUrl = getEmbedUrl(body, language);
-        });
-      });
 
       it('should be rendered', () => {
-        cy.get(`amp-iframe[src*="${embedUrl}"]`).should('be.visible');
-      });
+        cy.request(`${Cypress.env('currentPath')}.json`).then(
+          ({ body: jsonData }) => {
+            if (hasMedia(jsonData)) {
+              const embedUrl = getEmbedUrl(jsonData, language, true);
 
-      it('embed URL should be reachable', () => {
-        cy.testResponseCodeAndType(embedUrl, 200, 'text/html');
+              cy.get(`amp-iframe[src="${embedUrl}"]`).should('be.visible');
+              cy.testResponseCodeAndType(embedUrl, 200, 'text/html');
+
+              // Ensure media player is ready
+              cy.get('iframe').then($iframe => {
+                cy.wrap($iframe.prop('contentWindow'), {
+                  timeout: 30000,
+                })
+                  .its('embeddedMedia.playerInstances.mediaPlayer.ready')
+                  .should('eq', true);
+              });
+            } else {
+              cy.log(`No media on ${pageType} for ${service}`);
+            }
+          },
+        );
       });
     });
 
     describe('Chartbeat', () => {
-      if (envConfig.chartbeatEnabled) {
```

Comment:
We have a toggle for chartbeat - if we have to switch it off on the Test environment for any reason, this check may fail for AMP MAPs since it'd be expecting chartbeat to still be on the page.

Reply:
It's not possible to check for Chartbeat in Integration Tests, as it is rendered on client side. Shall I reinstate the `envConfig.chartbeatEnabled` instead?

Reply:
That chartbeat test literally only checks that the JSON inside an 'amp-analytics' tag has the text '50924' (our Chartbeat UID) and that it has been added to the page. It does no checking of functionality.